nautobot/extras/tests/test_models.py

```diff
@@ -1072,6 +1073,36 @@ def test_remote_url_hostname(self):
         self.repo.remote_url = "http://some-private-host/example.git"
         self.repo.validated_save()
 
+    def test_clone_and_discard_clone_functions(self):
+        """Confirm that the clone() and discard_clone() methods work as expected."""
+        self.repo.remote_url = "https://github.com/nautobot/nautobot.git"
+
+        with self.subTest("Clone a repository with no path argument provided"):
+            path = self.repo.clone()
+            # assert that the temporary directory was created in the expected location i.e. /tmp/
+            default_dir = "tmp"
+            self.assertIn(default_dir, path)
+            self.assertTrue(os.path.exists(path))
+            # assert that the temporary directory was discarded
+            self.assertTrue(self.repo.discard_clone(path))
+            self.assertFalse(os.path.exists(path))
+
+        with self.subTest("Clone a repository with a path argument provided"):
+            specified_path = "/source/nautobot/"
```

Comment:
This is specific to the docker-compose dev environment, and runs the risk of polluting your local source tree if the test errors out. Would be better to use `tempfile.mkdtemp()` to create a temp directory to use as the `specified_path` and clean it up afterwards.